bag_register/src/main/scala/uk/ac/wellcome/platform/archive/bag_register/services/Register.scala

```diff
@@ -19,13 +19,16 @@ import uk.ac.wellcome.platform.archive.common.storage.services.{
   BadFetchLocationException,
   StorageManifestService
 }
+import uk.ac.wellcome.storage.{Location, ObjectLocationPrefix, Prefix}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class Register(
-  bagReader: BagReader,
+class Register[BagLocation <: Location, BagPrefix <: Prefix[BagLocation]](
```

Comment:
Must we have the type bounds here, or does it work with just the base type?

Reply:
I think so, because otherwise the compiler can't be sure the type bounds on BagReader are satisfied.